sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py

```diff
@@ -0,0 +1,621 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+import time
+import logging
+import functools
+import uuid
+from typing import Any, List, TYPE_CHECKING, Optional, Union
+import six
+
+from uamqp import ReceiveClient, Source, types
+from uamqp.constants import SenderSettleMode
+
+from ._base_handler import BaseHandler
+from ._common.utils import create_authentication, utc_from_timestamp, utc_now
+from ._common.message import PeekMessage, ReceivedMessage
+from ._common.constants import (
+    REQUEST_RESPONSE_GET_SESSION_STATE_OPERATION,
+    REQUEST_RESPONSE_SET_SESSION_STATE_OPERATION,
+    REQUEST_RESPONSE_RENEW_SESSION_LOCK_OPERATION,
+    REQUEST_RESPONSE_RECEIVE_BY_SEQUENCE_NUMBER,
+    REQUEST_RESPONSE_UPDATE_DISPOSTION_OPERATION,
+    REQUEST_RESPONSE_RENEWLOCK_OPERATION,
+    REQUEST_RESPONSE_PEEK_OPERATION,
+    ReceiveSettleMode,
+    NEXT_AVAILABLE,
+    SESSION_LOCKED_UNTIL,
+    DATETIMEOFFSET_EPOCH,
+    SESSION_FILTER
+)
+from .exceptions import (
+    _ServiceBusErrorPolicy,
+    SessionLockExpired
+)
+from ._common import mgmt_handlers
+
+if TYPE_CHECKING:
+    import datetime
+    from azure.core.credentials import TokenCredential
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ServiceBusSession(object):
+    """
+    The ServiceBusSession is used for manage session states and lock renewal.
+
+    **Please use the instance variable `session` on the ServiceBusReceiver to get the corresponding ServiceBusSession
+    object linked with the receiver instead of instantiating a ServiceBusSession object directly.**
+
+    .. admonition:: Example:
+
+        .. literalinclude:: ../samples/sync_samples/sample_code_servicebus.py
+            :start-after: [START get_session_sync]
+            :end-before: [END get_session_sync]
+            :language: python
+            :dedent: 4
+            :caption: Get session from a receiver
+    """
+    def __init__(self, session_id, receiver, encoding="UTF-8"):
+        self._session_id = session_id
+        self._receiver = receiver
+        self._encoding = encoding
+        self._session_start = None
+        self._locked_until_utc = None
+        self.auto_renew_error = None
```

Comment:
Should this be public?

Reply:
This is inherited from Track1. Auto_renew feature is something not discussed yet, so we want to keep it the same as Track1 for preview1.